pennylane/operation.py

```diff
@@ -1164,10 +1181,12 @@ def grad_method(self):
         """
         if self.num_params == 0:
             return None
+        if self.grad_recipe != [None] * self.num_params:
+            return "A"
```

Comment:
Oh, this is basically checking if `grad_recipe` has been defined?

Reply:
Yes exactly. If users/devs provide a grad_recipe, we know that `grad_method` should be `"A"`, even if they didn't say so, right?

Reply:
yep!